src/main/java/io/anserini/document/TrecTextRecord.java

```diff
@@ -0,0 +1,74 @@
+package io.anserini.document;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.anserini.index.IndexWebCollection;
+
+public final class TrecTextRecord {
+
+  public static final String DOCNO = "<DOCNO>";
+  public static final String TERMINATING_DOCNO = "</DOCNO>";
+
+  public static final String DOC = "<DOC>";
+  public static final String TERMINATING_DOC = "</DOC>";
+
+  public static final String[] startTags = {"<TEXT>", "<HEADLINE>", "<TITLE>", "<HL>", "<HEAD>",
+          "<TTL>", "<DD>", "<DATE>", "<LP>", "<LEADPARA>"
+  };
+  public static final String[] endTags = {"</TEXT>", "</HEADLINE>", "</TITLE>", "</HL>", "</HEAD>",
+          "</TTL>", "</DD>", "</DATE>", "</LP>", "</LEADPARA>"
+  };
+
+  public static final int BUFFER_SIZE = 1 << 16; // 64K
+
+  public static WarcRecord parseTrecTextRecord(StringBuilder builder) {
```

Comment:
I agree that this is confusing...
The reason probably is that indexing TrecRecord is exactly the same as indexing the WarcRecord after we get the contents. Trec record includes tags such as <TEXT> <HEADLINE> etc. We need jsoup to extract the pure text out of it. 
If I create another record -- TrecRecord, I will duplicate the indexWarcRecord for indexing the Trec record by only changing the function args. 
I think the better way to deal with this is we refactor the code so that different types of the records inherit a parent class. That means a lot of renaming though...
What do you think?